src/ai/backend/manager/idle.py

```diff
@@ -343,13 +370,29 @@ async def check_session(self, session: Row, dbconn: SAConnection) -> bool:
         return False
 
 
+class SessionLifetimeChecker(BaseIdleChecker):
+
+    name: ClassVar[str] = "session_lifetime"
+
+    async def populate_config(self, config: Mapping[str, Any]) -> None:
+        pass
+
+    async def check_session(self, session: Row, dbconn: SAConnection, policy: Row) -> bool:
+        now = await dbconn.execute(sa.select(sa.func.now()))
+        if policy["max_session_lifetime"] >= 0:
+            # TODO: once per-status time tracking is implemented, let's change created_at
+            #       to the timestamp when the session entered PREPARING status.
+            if now - session["created_at"] >= policy["max_session_lifetime"]:
```

Comment:
If `max_session_lifetime` is zero, I suggest interpreting it as infinity. Also, these lines raise exceptions due to calculation type-mismatch issues. These might be updated to something like below:

```python
now = await dbconn.scalar(sa.select(sa.func.now()))
if policy["max_session_lifetime"] > 0:
    # TODO: once per-status time tracking is implemented, let's change created_at
    #       to the timestamp when the session entered PREPARING status.
    if now - session["created_at"] >= timedelta(seconds=policy["max_session_lifetime"]):
```